### ISSUE TITLE
SSL error after AWS Aurora Update

### ISSUE THREAD

**Issue by HyperBrain** [mysqljs/mysql#1554]
After upgrading our AWS Aurora servers today (us-east-1 region), we cannot connect anymore using SSL: "Amazon RDS".

In my opinion the RDS SSL certificate in https://github.com/mysqljs/mysql/blob/96fdd0566b654436624e2375c7b6604b1f50f825/lib/protocol/constants/ssl_profiles.js has to be updated.

**Comment by dougwilson** [comment]
That's interesting. I can certainly update them (or you can even make a pull request to update). Can you link me to the public certificate that I need to add/update?

**Comment by HyperBrain** [comment]
We will try to download the new RDS certificates and patch our local mySQL NPM to see if it works.
I will comment the outcome here and we'll either prepare a PR or supply the certificates.

**Comment by HyperBrain** [comment]
**Ok. I got answer from AWS. It is not a problem with the certificates - so node mysql is ok!**

In the case that anyone else runs into the problem:
The AWS Aurora patch is broken, but there is a temporary workaround

"The temporary workaround is to disable Diffie-Hellman ciphers suites in the impacted MySQL client.
mysql -hyourhost -uyouruser -p{} --ssl-ca=rds-combined-ca-bundle.pem --ssl-verify-server-cert --ssl-cipher='DEFAULT:!EDH:!DHE'"

**Comment by valeshino** [comment]
We were also having ssl handshake errors when connecting to an Amazon RDS Aurora backend using the "Amazon RDS" ssl config in this module. The errors we were getting were:
```json
{
      "code": "HANDSHAKE_SSL_ERROR",
      "message": "140283231541056:error:14082174:SSL routines:ssl3_check_cert_and_algorithm:dh key too small:../deps/openssl/openssl/ssl/s3_clnt.c:3617:"
}
```

This seems to coincide with an [update to the Aurora engine](http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Aurora.DatabaseEngineUpdates.20161026.html) on 10-26:

The fix was relatively simple (using @HyperBrain's response from AWS), but less than obvious, especially if you're less adept at SSL in general. We downloaded the [AWS combined certificate](https://s3.amazonaws.com/rds-downloads/rds-combined-ca-bundle.pem) into our repository and update our ssl config as:

```javascript
const fs = require('fs');
const amazonCA = fs.readFileSync('./rds-combined-ca-bundle.pem');
const sslConfig = {
    ca: amazonCA,
    ciphers: 'DEFAULT:!EDH:!DHE', // from @HyperBrain's AWS response
};

var connection = mysql.createConnection({
    host: '...',
    ssl : sslConfig,
});
```

This workaround is fine for now and we'll work with AWS to come up with a more permanent solution and update the issue / file a PR if it's something that can be addressed directly in this module.

**Comment by dougwilson** [comment]
> and update the issue / file a PR if it's something that can be addressed directly in this module

Please do!

**Comment by valeshino** [comment]
Since the workaround was working, we didn't focus too much on the issue until today and I had it slightly wrong. The [10-26 update](http://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Aurora.DatabaseEngineUpdates.20161026.html) actually *fixed* the issue. It's only a problem on the 1.8 (fixed in 1.8.1, broken in 1.8) version of the aurora engine, so the easy fix is to just update your aurora instances. If that's easier said than done, the workaround above is sufficient, but I don't want to change the module and add complexity that's only needed for one patch revision of the aurora engine for RDS.
